src/main/kotlin/com/depromeet/piki/admin/announcement/AdminAnnouncementController.kt

```
@@ -41,22 +44,94 @@ class AdminAnnouncementController(
-            return "redirect:/admin/announcements?error=length"
+        val safePushTitle = pushTitle ?: ""
+        val safePushBody = pushBody ?: ""
+        // 검증·rehost 실패로 edit 로 되돌릴 땐 방금 입력한 값을 flash 로 보존해 재주입한다 — 긴 패치노트를 다시 쓰게 하지 않는다.
+        if (!validLengths(title, safeBody, safePushTitle, safePushBody)) {
+            preserveDraftInput(redirectAttributes, title, safeBody, pushEnabled, safePushTitle, safePushBody)
+            return "redirect:/admin/announcements/$id/edit?error=length"
+        }
+        return try {
+            adminAnnouncementService.update(id, title, safeBody, pushEnabled, safePushTitle, safePushBody, actor(request), clientIp(request))
+            "redirect:/admin/announcements/$id/send"
+        } catch (e: AnnouncementImageException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)